arbitrum/backend.go

```diff
@@ -67,10 +70,57 @@ func (b *Backend) ChainDb() ethdb.Database {
 	return b.chainDb
 }
 
+type Transaction struct {
+	Vm      uint64 `json:"vm"`
+	Payload []int  `json:"payload"`
+}
+
 func (b *Backend) EnqueueL2Message(ctx context.Context, tx *types.Transaction, options *arbitrum_types.ConditionalOptions) error {
+	if b.config.Espresso {
+		return b.EnqueueL2MessageEspresso(ctx, tx, options)
+	}
+	return b.EnqueueL2MessageDefault(ctx, tx, options)
+}
+
+func (b *Backend) EnqueueL2MessageDefault(ctx context.Context, tx *types.Transaction, options *arbitrum_types.ConditionalOptions) error {
 	return b.arb.PublishTransaction(ctx, tx, options)
 }
 
+func (b *Backend) EnqueueL2MessageEspresso(ctx context.Context, tx *types.Transaction, options *arbitrum_types.ConditionalOptions) error {
+	var txnBytes, err = json.Marshal(tx)
+	if err != nil {
+		return err
+	}
+	//	json.RawMessage is a []byte array, which is marshalled as a base64-encoded string.
+	//	Our sequencer API expects a JSON array.
+	payload := make([]int, len(txnBytes))
+	for i := range payload {
+		payload[i] = int(txnBytes[i])
+	}
+	txn := Transaction{
+		Vm:      b.config.EspressoNamespace,
+		Payload: payload,
+	}
+	marshalled, err := json.Marshal(txn)
+	if err != nil {
+		return err
+	}
+	request, err := http.NewRequest("POST", b.config.HotShotUrl, bytes.NewBuffer(marshalled))
+	if err != nil {
+		return err
+	}
+	request.Header.Set("Content-Type", "application/json")
+	client := &http.Client{}
+	response, err := client.Do(request)
+	if err != nil {
+		return err
+	}
```

Comment:
I think we should close response bodies if `err != nil` (?) https://pkg.go.dev/net/http
```suggestion
	}
	defer response.Body.Close()
```